src/spiffe_id/mod.rs

```diff
@@ -189,18 +169,65 @@ impl TryFrom<String> for SpiffeId {
 impl TryFrom<&str> for SpiffeId {
     type Error = SpiffeIdError;
     fn try_from(s: &str) -> Result<SpiffeId, Self::Error> {
-        Self::from_str(s)
+        Self::new(s)
+    }
+}
+
+// Validates that a path string is a conformant path for a SPIFFE ID. Namely:
+// - does not contain an empty segments (including a trailing slash)
+// - does not contain dot segments (i.e. '.' or '..')
+// - does not contain any percent encoded characters
+// - has only characters from the unreserved or sub-delims set from RFC3986.
+fn validate_path(path: &str) -> Result<(), SpiffeIdError> {
+    if path.is_empty() {
+        return Ok(());
+    }
+
+    if (path.as_bytes()[0] as char) != '/' {
```

Comment:
Maybe `path.chars().next()` is clearer?
Since `as_bytes` is used I am wondering what says SPIFFE about non-ASCII characters being used in SPIFFE IDs?

Reply:
Good catch! I replaced it with `!path.starts_with('/')`, I shouldn't have used `as_bytes`, non-ASCII characters are not allowed.